libs/blocks/merch/merch.js

```diff
@@ -116,17 +124,62 @@ export async function getPriceContext(el, params) {
   };
 }
 
+let loadEntitlementsPromise;
+
+export async function loadEntitlements() {
+  await loadIms();
+  if (!window.adobeIMS?.isSignedInUser?.()) return Promise.resolve([{}, {}]);
+  const [{ default: getUserEntitlements },
+    mappings] = await Promise.all([
+    import('../global-navigation/utilities/getUserEntitlements.js'),
+    fetch(entitlementsURL),
+  ]);
+  const entitlements = await getUserEntitlements();
+  const entitlementsMetadata = await mappings.json();
```

Comment:
Did you have a look whether grouping these into a `Promise.all` might have any benefit?

Reply:
given that the fetch is already done, I don't expect big difference, but I can still do it.